### PR TITLE
spelling: Replace occurrences of "iff" with "if and only if"

### PR DESCRIPTION
**Pull request by pdgendt**
Spell checking tools do not recognize "iff", replace with "if and only if".
See https://en.wikipedia.org/wiki/If_and_only_if

### CODE REVIEW

**Review by nashif — DISMISSED** [review]

**Comment by ajf58** [comment]
Is this definitely a typo? Authors might be using this coming from a more academic background, where  'if and only if'[1] is meant here as a more restrictive conditional 'if'.

[1] https://en.wikipedia.org/wiki/If_and_only_if

**Comment by pdgendt** [comment]
> Is this definitely a typo? Authors might be using this coming from a more academic background, where 'if and only if'[1] is meant here as a more restrictive conditional 'if'.
> 
> [1] https://en.wikipedia.org/wiki/If_and_only_if

The more you know... It does make it harder for spellchecker tools as it isn't recognized.

File: dts/bindings/base/mutable.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ properties:
   zephyr,mutable:
     type: boolean
     description: |
-      True iff the device structure may be mutated.
+      True if the device structure may be mutated.
```
Comment (pdgendt) [inline dts/bindings/base/mutable.yaml:10]: @cfriedt was the "iff" intentional here? :)

Comment (cfriedt) [reply dts/bindings/base/mutable.yaml:10]: Yes - please replace with "if and only if"

**Review by cfriedt — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by kartben — APPROVED** [review]
looks like there might be a few more? (e.g. `include/zephyr/xen/public/arch-arm.h`, `include/zephyr/net/http/parser_url.h`, ...)

https://github.com/search?q=repo%3Azephyrproject-rtos%2Fzephyr+%22+iff+%22&type=code

**Comment by pdgendt** [comment]
> looks like there might be a few more? (e.g. `include/zephyr/xen/public/arch-arm.h`, `include/zephyr/net/http/parser_url.h`, ...)
> 
> https://github.com/search?q=repo%3Azephyrproject-rtos%2Fzephyr+%22+iff+%22&type=code

I didn't modify those, as these files appear to be copies of MIT projects.

**Review by jhedberg — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Pull request merged**